### ISSUE TITLE
Initialising a new Open mSupply site with a lot of central server data takes a long time

### ISSUE THREAD

**Issue by DhanyaHerath** [msupply-foundation/open-msupply#4658]
## What went wrong? 😲
From Mark G: 

> Why does an Open mSupply store which:
> 
> * Handles items from one master list containing a dozen items
> * Receives stock from 1 supplier
> * Issues to 6 non-store customers
> Needs to be initialised with several hundred thousand records?  This takes literally hours at the end of a the average  cellular phone line, and is impractically long in the field.  The current work-around is to configuring devices where the internet is better than average, and then take them to the field.  We won't get away with this forever.  The time will come when we need to initiate a site in the field.
> 
> Why must every database have all system data (every item, master list, non-patient name, item-store-join, etc.)?  Why can't these details be synced as visibility to them is turned on?
> 

## How to Reproduce 🔨

Steps to reproduce the behaviour:

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

**Comment by andreievg** [comment]
Triage, setting priority as must have to priorities refinement

**Comment by andreievg** [comment]
Needs investigation, what is the breakdown of sync outs for small new sites.
The goal is to try and reduce number of records needed to be sent so let's see if we can identify any low hanging fruit.
Also discussed trying out connection timeout in reqwest crate and maybe do retries of connection.

**Comment by andreievg** [comment]
During dev test day, me and @lache-melvin had issues initialising tablets, it was very slow and sometimes connection would be stuck for a long time, only to show up with `retry`. I did some investigation and even put caddy between mSupply and the outside world to setup timeouts, this didn't really have any effect. I added logging and saw that when connection is being 'timed' out, there wasn't even any caddy logs present (so was outside control of central servers).

There is a way to set connection timeout for reqwest, using client builder, while I was trying to set timeouts and replies I stumbled accros gzip header, in reqwest docs looks like it's not automatically turned on, and need gzip features for it to be enabled: https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.gzip

[In this branch](https://github.com/msupply-foundation/open-msupply/compare/4662-omSupply-backups...Checking-sync-connection) I experimented with gzip, connection timeout and retries, and batch size increase. I did a few benchmarks connecting to the same central servers on debug build apk and it looks like it's 100-150% faster. I don't think the timeout and connection is helping, because today I never got the 'retry' msg.

**Conclusions**

* Add hotfix for 2.2 with `gzip`
* Send test APK with `gzip` and connection retry to `Mark G` for someone to try in country (it would need to be 1.7 I think), if there is significant improvement then we should add batch size settings and connection timeout/retry setting on initialisation screen.
* Increase batch size on mSupply end to 5k
* Add issue for refactor week to check dynamic/optimal batch size detection, we've done this for mobile, we've got expert for this @Chris-Petty
